src/scheduler.rs

```diff
@@ -596,20 +369,20 @@ impl RoleGroupEligibleNodes {
     fn preferred_node_or<F>(
         &self,
         pod: &PodIdentity,
-        preferred: Option<&NodeIdentity>,
+        preferred: Option<NodeIdentity>,
         default: F,
     ) -> Option<NodeIdentity>
     where
         F: Fn(&PodIdentity, &Vec<NodeIdentity>) -> Option<NodeIdentity>,
     {
         match self
             .node_set
-            .get(&pod.role)
-            .and_then(|role| role.get(&pod.group))
+            .get(&pod.role().to_string())
+            .and_then(|role| role.get(&pod.group().to_string()))
         {
             Some(nodes) if !nodes.is_empty() => {
                 if let Some(node_id) = preferred {
-                    let tmp = nodes.iter().find(|n| *n == node_id);
+                    let tmp = nodes.iter().find(|n| n == &&node_id);
```

Comment:
double reference? I think one is enough (not tested).

Reply:
Clippy says "no".